### PR TITLE
Guard when calling AfterScenario when TestRunner is already released

### LINKED ISSUE #387: Observing System.InvalidOperationException : TestThreadContext with id  was already released
### Reqnroll Version

2.2.1

### Which test runner are you using?

xUnit

### Test Runner Version Number

2.9.0

### .NET Implementation

.NET 8.0

### Test Execution Method

Visual Studio Test Explorer

### Content of reqnroll.json configuration file

_No response_

### Issue Description

We have recently migrated from Specflow and facing issue with test execution.

The test fails with following error (StackTrace)

Stack Trace: 
TestRunnerManager.ReleaseTestThreadContext(ITestThreadContext testThreadContext)
TestRunnerManager.ReleaseTestRunner(ITestRunner testRunner)
**StatusCheckFeature.TestTearDownAsync()** // our feature file
IAsyncLifetime.DisposeAsync()

Following versions are working:

- 2.0.2 
- 2.0.3

Following versions having the issue
- 2.2.1
- 2.2.0
- 2.1.1
- 2.1.0

Could anyone assist on the issue?

### Steps to Reproduce

Run the test from test explorer. 

### Link to Repro Project

_No response_

### PR DESCRIPTION
**Pull request by obligaron**
### 🤔 What's changed?

Checks in generated feature files that `testRunner` is not null in after scenario hooks (IAsyncLifetime.DisposeAsync/TearDown/TestCleanup). If the testRunner field is null return without a exception.

### ⚡️ What's your motivation? 

Hopefully fixes #387

This PR should fix crashes caused by the test framework entering the after scenario hooks multiple times.
If Reqnroll has incorrect TestRunner management, this PR won't fix it.
See also discussion in #550.

Unfortunately, we don't have a sample project that reproduces #387.

### 🏷️ What kind of change is this?

- :bug: Bug fix (non-breaking change which fixes a defect)

### ♻️ Anything particular you want feedback on?

Is this the right approach?
Should we add some sort of logging for this (not expected) behavior of the test framework?

### 📋 Checklist:

- [x] Users should know about my change
  - [x] I have added an entry to the "[vNext]" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request & included my GitHub handle to the release contributors list.

----

*This text was originally taken from the [template of the Cucumber project](https://github.com/cucumber/.github/blob/main/.github/PULL_REQUEST_TEMPLATE.md), then edited by hand. [You can modify the template here.](https://github.com/reqnroll/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs [hunk 1]
```diff
@@ -385,6 +385,19 @@ private void SetupTestCleanupMethod(TestClassGenerationContext generationContext
 
             var testRunnerField = _scenarioPartHelper.GetTestRunnerExpression();
 
+            // Guard when calling AfterScenario when TestRunner is already released (errors or edge cases in a supported test framework)
+            // if ((testRunner == null))
```
Comment (304NotModified) [inline Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs:389]: what's the idea with this commented out code?

File: Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs [hunk 2]
```diff
@@ -393,19 +406,23 @@ private void SetupTestCleanupMethod(TestClassGenerationContext generationContext
             var onScenarioEndCallStatement = new CodeExpressionStatement(onScenarioEndCallExpression);
 
             // "Release" the TestRunner, so that other threads can pick it up
-            // TestRunnerManager.ReleaseTestRunner(testRunner);
+            // global::Reqnroll.TestRunnerManager.ReleaseTestRunner(testRunner);
```
Comment (304NotModified) [inline Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs:409]: Same for this

File: Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs [hunk 3]
```diff
@@ -393,19 +406,23 @@ private void SetupTestCleanupMethod(TestClassGenerationContext generationContext
...
             var releaseTestRunnerCallStatement = new CodeExpressionStatement(
                 new CodeMethodInvokeExpression(
                     new CodeTypeReferenceExpression(new CodeTypeReference(typeof(TestRunnerManager), CodeTypeReferenceOptions.GlobalReference)),
                     nameof(TestRunnerManager.ReleaseTestRunner),
                     testRunnerField));
 
+            // Unassign TestRunner to make sure it won't be reused if the test framework runs AfterScenario multiple times
+            // testRunner = null;
```
Comment (304NotModified) [inline Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs:417]: Same

File: Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs (hunk 1 above)
Comment (clrudolphi) [reply Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs:389]: In the generator code, the commented lines demonstrate the code that will get generated by the CodeDom code that follows the comment.

Comment (304NotModified) [reply Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs:389]: Maybe we could prefix the comments with: `generates:`

**Review by gasparnagy — APPROVED** [review]
@obligaron I think this is useful, so let's go for it.

@304NotModified Your comment about the commented generated code makes sense, but we use this style of comments all across the generator project, so it does not make sense to change the comment style for this particular PR. If we want to make a clearer distinction of the real comments and the illustration of the generated code, we should do that as a separate PR.
